Release 0.6 (#205)
NREL/PVDegradationTools: docs/source/whatsnew/releases/v0.6.0.rst

  ``False`` to avoid unnecessary network calls when reading local files. (:pull:`223`)
- Add `"name"`, `"units"`, and `"value"` fields to the materials JSON databases
  `AApermeation.json`, `H2Opermeation.json`, and `O2permeation.json`.
  (:issue:`211`, :pull:`224`)
- Add new (public) function :py:mod:`pvdeg.humidity` to calculate relative humidity from
  air temperature and dew point temperature. (:issue:`227`, :pull:`228`)

Deprecations
-------------
- ``GeospatialScenario._get_geospatial_data()``
- ``GeospatialScenario.getValidRegions()``
- ``GeospatialScenario.gids_tonumpy()``
- ``GeospatialScenario.gids_tolist()``
- Deprecate `see_added` argument, and associated UserWarning for succesful addition of
  objects, in several :py:mod:`~pvdeg.scenario` functions. (:pull:`192`)

Bug Fixes
---------
- ``v0.5.0`` incorrectly claimed that ``GeospatialScenario`` was moved to
  ``pvdeg/geospatialscenario.py``. Completed the move.
- Code cleanup and standardization in :py:mod:`pvdeg.degradation` module:
  - Improve readability of mathematical expressions
  - Refactor redundant helper functions
  - Update docstrings
  - Remove unused functions and variables
  - Improve code consistency

>   (:issue:`177`, :pull:`178`)

Reviewer: I think this should be moved to the enhancements section

v0.6.0 (2025-08-13)
===================

Breaking Changes
----------------
- Update dependency requirements:
  - pvlib-python: was ==0.10.3, now >=0.12.0
  - numpy: was ==1.26.4, now >1.19.3
  - scipy: was <1.14.0, now >1.6.0
  (:issue:`103`, :pull:`163`)
- Remove deprecated `vertical_poa` function from `pysam.py` and related (incorrect)
  reference to `standards.vertical_poa` in `geospatial.py`. (:issue:`167`, :pull:`174`)

Enhancements
------------

> - Changed ``GeospatialScenario.get_geospatial_data()`` to a ``@property``, now use ``GeospatialScenario.geospatial_data``
> - Changed ``GeospatialScenario.coords_tonumpy()`` to a ``@property``, now use ``GeospatialScenario.coords``
> - Refactor ``META_MAP`` in :py:mod:`pvdeg.weather` to module-level constant
>   (:pull:`182`)
> - Updated pytest workflow to cancel in-progress runs for the same branch in favour of
>   running on the latest commit. (:pull:`191`)
> - Added support for pandas dataframes to :py:mod:`pvdeg.weather.map_meta()` (:pull:`193`)

Reviewer: It is my understanding that the imperative form is best practice for changelogs

```suggestion
- Change ``GeospatialScenario.get_geospatial_data()`` to a ``@property``, now use ``GeospatialScenario.geospatial_data``
- Change ``GeospatialScenario.coords_tonumpy()`` to a ``@property``, now use ``GeospatialScenario.coords``
- Refactor ``META_MAP`` in :py:mod:`pvdeg.weather` to module-level constant
  (:pull:`182`)
- Update pytest workflow to cancel in-progress runs for the same branch in favour of
  running on the latest commit. (:pull:`191`)
- Add support for pandas dataframes to :py:mod:`pvdeg.weather.map_meta()` (:pull:`193`)

```

Author: done

> 

Reviewer: ```suggestion

- Add a new GitHub Actions workflow to automatically run flake8 for linting on pull
  requests (:issue:`164`, :pull:`165`)
- Lint codebase for PEP8 compliance (:pull:`166`, :pull:`175`, :pull:`196`)

```

Author: done

v0.6.0 (2025-08-13)
===================

Breaking Changes
----------------
- Update dependency requirements:
  - pvlib-python: was ==0.10.3, now >=0.12.0
  - numpy: was ==1.26.4, now >1.19.3
  - scipy: was <1.14.0, now >1.6.0
  (:issue:`103`, :pull:`163`)
- Remove deprecated `vertical_poa` function from `pysam.py` and related (incorrect)
  reference to `standards.vertical_poa` in `geospatial.py`. (:issue:`167`, :pull:`174`)
- Replaced ambiguous variable name `l` in
  :py:func`pvdeg.collection.collection_probability` with `diffusion_length` to enhance
  code readability and maintainability. (:issue:`198`, :pull:`199`)
- Replace print messages for failure or succesful addition of objects (location, job...)
  with try/except blocks and warnings in :py:mod:`~pvdeg.scenario`. (:pull:`192`)
- Remove deprecated `pvdeg.store` module. (:issue:`168`, :pull:`180`)

> 

Reviewer: I noticed this after digging through #134 to understand some of the notebook failures I am encountering while working on #184. I think we still need to investigate the changes in #134 further and document them carefully, especially breaking changes like this
```suggestion
- Rename `pvdeg.degradation.degradation` to :py:func:`pvdeg.degradation.spectral_degradation`.  (:pull:`134`)

```

Enhancements
------------
- Change ``GeospatialScenario.get_geospatial_data()`` to a ``@property``, now use
``GeospatialScenario.geospatial_data``
- Change ``GeospatialScenario.coords_tonumpy()`` to a ``@property``, now use
``GeospatialScenario.coords``
- Refactor ``META_MAP`` in :py:mod:`pvdeg.weather` to module-level constant
  (:pull:`182`)
- Update pytest workflow to cancel in-progress runs for the same branch in favour of
  running on the latest commit. (:pull:`191`)
- Add support for pandas dataframes to :py:mod:`pvdeg.weather.map_meta()` (:pull:`193`)
- Add a new GitHub Actions workflow to automatically run flake8 for linting on pull
  requests (:issue:`164`, :pull:`165`)
- Lint codebase for PEP8 compliance (:pull:`166`, :pull:`175`, :pull:`196`)

Deprecations
-------------
- ``GeospatialScenario._get_geospatial_data()``
- ``GeospatialScenario.getValidRegions()``
- ``GeospatialScenario.gids_tonumpy()``
- ``GeospatialScenario.gids_tolist()``
- Deprecate `see_added` argument, and associated UserWarning for succesful addition of

>   objects, in several :py:mod:`~pvdeg.scenario` functions. (:pull:`192`)

Reviewer: Might be more accurate to change this to "remove" and move it into the breaking changes section
I did not follow a deprecation cycle with #192. I have renamed the PR accordingly.
